lib/mutant/result.rb

```diff
@@ -206,25 +245,39 @@ class Mutation
         :runtime
       )
 
+      # Create mutation criteria results
+      #
+      # @praam [Result::CoverageCriteria]
+      def criteria_result(coverage_criteria)
+        CoverageCriteria.new(
+          test_result: coverage_criteria.test_result && test_result_success?,
+          timeout:     coverage_criteria.timeout     && timeout?
+        )
+      end
+
       # Time the tests had been running
       #
       # @return [Float]
       def killtime
-        if isolation_result.success?
-          isolation_result.value.runtime
-        else
-          0.0
-        end
+        isolation_result.value&.runtime || 0.0
```

Comment:
Its now perfectly valid there is no value returned from the isolation. And the status of the islation result cannot be determined anymore without the coverage criteria coming from the outside.